Automattic/wp-calypso
# Add text domain to copy in domain-picker and plans-grid packages

#### Changes proposed in this Pull Request

* Define `__i18n_text_domain__`
* Hard code `__i18n_text_domain__` to `'default'` in calypso
* Hard code `__i18n_text_domain__` to `'full-site-editing'` in ETK
* Hard code `__i18n_text_domain__` to some value in the unit tests so they don't fail
* Switch `@automattic/domain-picker` and `@automattic/plans-grid` over to use `@wordpress/18n`
* Add the text-domain constant to all calls to `__()` in the `domain-picker` and `plans-grid` packages

#### Testing instructions



* Check out branch and test using `yarn dev --sync`
* Switch user to Spanish or something non-English
* Go to `/new/domains` and see that the "View more results" button and other copy is still translated correctly
* Open the page editor on an unlaunched gutenboarding site
* Click "Complete setup" in the top right
* Click the second step in the launch flow, which is for choosing a domain
  * The text in here is being translated, but will probably still appear as English because translations aren't done for the ETK (the strings are translated for the default text-domain, but not the full-site-editing text-domain)
  * You can confirm the translations are working by switching one of the strings in the domain picker with something that we know is already translated in the full-site-editing text domain
  * e.g. Switch "Search for a domain" with "Complete setup", rebuild, and the domain search placeholder text should be translated.
